src/sortStudents.ts

```diff
@@ -1,16 +1,77 @@
 
 export interface Student {
-  // describe Student interface
+  name: string;
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
+type StudentKey = keyof Student;
+
 export enum SortType {
-  // describe SortType enum
+  Name,
+  Surname,
+  Age,
+  Married,
+  AverageGrade,
 }
 
 // create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
+
+function GetAverageValue(array: []): number {
+  return array.reduce((sum, value) => sum + value, 0) / array.length;
+}
+
+function defineValue(value: StudentKey): StudentKey | number {
+  return Array.isArray(value)
+    ? GetAverageValue(value)
+    : value;
+}
+
+function compareStudentsByKey(
+  key: StudentKey,
+  order: SortOrder,
+): (currentStudent: Student, nextStudent: Student) => number {
+  return function callback(
+    currentStudent: Student,
+    nextStudent: Student,
+  ): number {
+    const currentValue = defineValue(currentStudent[key]);
+    const nextValue = defineValue(nextStudent[key]);
+
+    const comparison = typeof currentValue === 'string'
+      ? currentValue.toString().localeCompare(nextValue.toString())
+      : Number(currentValue) - Number(nextValue);
+
+    return order === 'asc' ? comparison : -comparison;
+  };
+}
+
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+) : Student[] {
+  const studentsCopy = [...students];
+
+  switch (sortBy) {
+    case SortType.Name:
+      return studentsCopy.sort(compareStudentsByKey('name', order));
```

Comment:
```suggestion
      return studentsCopy.sort(compareStudentsByKey(SortType.name, order));
```
use SortType to not hardcode keys.